src/test/java/ui/base/UITestBase.java

```diff
@@ -0,0 +1,63 @@
+package ui.base;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Optional;
+import org.testng.annotations.Parameters;
+
+import static org.realworld.api.Constants.UI_BASE_URL;
+
+public class UITestBase {
+
+    private WebDriver driver;
```

Comment:
**major**: this field is redundant, the driver already exists in the thread local. More over, the driver is shared between threads, so if driver exists in the driver var, it doesn't mean it exist in the thread local for the current thread.

Reply:
done